pkgs/tools/text/enscript/0001-use-system-getopt.patch

```diff
@@ -0,0 +1,1412 @@
+From 816887498bf18c987cfec9015ca7d74700b7dda4 Mon Sep 17 00:00:00 2001
+From: annalee <150648636+a-n-n-a-l-e-e@users.noreply.github.com>
+Date: Sun, 3 Dec 2023 04:26:25 +0000
+Subject: [PATCH 1/2] use system getopt
+
+---
+ compat/Makefile.am |    4 +-
+ compat/Makefile.in |    6 +-
+ compat/getopt.c    | 1000 --------------------------------------------
+ compat/getopt.h    |  133 ------
+ compat/getopt1.c   |  189 ---------
+ 5 files changed, 5 insertions(+), 1327 deletions(-)
+ delete mode 100644 compat/getopt.c
+ delete mode 100644 compat/getopt.h
+ delete mode 100644 compat/getopt1.c
+
+diff --git a/compat/Makefile.am b/compat/Makefile.am
+index aefc06f..23aaf81 100644
+--- a/compat/Makefile.am
++++ b/compat/Makefile.am
+@@ -25,7 +25,7 @@
+ AUTOMAKE_OPTIONS = no-dependencies
+ 
+ noinst_LIBRARIES = libcompat.a
+-libcompat_a_SOURCES = getopt.c getopt1.c xalloc.c regex.c
++libcompat_a_SOURCES = xalloc.c regex.c
+ libcompat_a_LIBADD = @LIBOBJS@ @ALLOCA@
+ libcompat_a_DEPENDENCIES = @LIBOBJS@ @ALLOCA@
+ 
+@@ -34,7 +34,7 @@ libcompat_a_SOURCES_windelta =	\
+ 	+../w32/getpwd.c	\
+ 	+../w32/getuid.c
+ 
+-noinst_HEADERS = getopt.h regex.h xalloc.h
++noinst_HEADERS = regex.h xalloc.h
+ 
+ EXTRA_DIST = ChangeLog.old strerror.c memmove.c memcpy.c strtol.c	\
+ 	strtoul.c gettext.h
+diff --git a/compat/Makefile.in b/compat/Makefile.in
+index edfc620..dee05d7 100644
+--- a/compat/Makefile.in
++++ b/compat/Makefile.in
+@@ -137,7 +137,7 @@ am__v_AR_ = $(am__v_AR_@AM_DEFAULT_V@)
+ am__v_AR_0 = @echo "  AR      " $@;
+ am__v_AR_1 = 
+ libcompat_a_AR = $(AR) $(ARFLAGS)
+-am_libcompat_a_OBJECTS = getopt.$(OBJEXT) getopt1.$(OBJEXT) \
++am_libcompat_a_OBJECTS = \
+ 	xalloc.$(OBJEXT) regex.$(OBJEXT)
+ libcompat_a_OBJECTS = $(am_libcompat_a_OBJECTS)
+ AM_V_P = $(am__v_P_@AM_V@)
+@@ -326,7 +326,7 @@ top_builddir = @top_builddir@
+ top_srcdir = @top_srcdir@
+ AUTOMAKE_OPTIONS = no-dependencies
+ noinst_LIBRARIES = libcompat.a
+-libcompat_a_SOURCES = getopt.c getopt1.c xalloc.c regex.c
++libcompat_a_SOURCES = xalloc.c regex.c
+ libcompat_a_LIBADD = @LIBOBJS@ @ALLOCA@
+ libcompat_a_DEPENDENCIES = @LIBOBJS@ @ALLOCA@
+ libcompat_a_SOURCES_windelta = \
+@@ -334,7 +334,7 @@ libcompat_a_SOURCES_windelta = \
+ 	+../w32/getpwd.c	\
+ 	+../w32/getuid.c
+ 
+-noinst_HEADERS = getopt.h regex.h xalloc.h
++noinst_HEADERS = regex.h xalloc.h
+ EXTRA_DIST = ChangeLog.old strerror.c memmove.c memcpy.c strtol.c	\
+ 	strtoul.c gettext.h
+ 
+diff --git a/compat/getopt.c b/compat/getopt.c
+deleted file mode 100644
+index 752f28a..0000000
+--- a/compat/getopt.c
```

Comment:
Maybe we should just rm those files to not have potential Behr conflicts on updates.

Reply:
maybe but the files haven't changed in over 20 years and the patch keeps everything contained in the same place. doesn't really see to matter either way so i am going to leave it as is.

Reply:
It makes the patch self contained yes, but adding ~1400 lines of code is not IMO worth it. Consider asserting that the files remain unchaned with something like this in postPatch:

```bash
test "$(sha256sum "the-file-in-question" | cut -d' ' -f1)" = "03127f8b3bc73ec1f921557131ee3ecd4e431191948970677d674b4acc99c4fe"
```

Reply:
i disagree but changed the code to just remove the conflicting header in the postPatch.